homeassistant/components/climate/radiotherm.py

```diff
@@ -137,23 +137,43 @@ def is_away_mode_on(self):
 
     def update(self):
         """Update the data from the thermostat."""
-        self._current_temperature = self.device.temp['raw']
+        current_temp = self.device.temp['raw']
+        while current_temp == -1:
+            _LOGGER.error("Invalid temperature detected, retrying")
+            current_temp = self.device.temp['raw']
```

Comment:
Does `self.device.temp['raw']` do I/O and actually fetch the temperature ?

Also, keep hammering the thermostat indefinitely might not be a good idea. Better to limit it to 5 retries and peace out otherwise.